Modules/Bridge/VTK/include/itkVTKImageImport.hxx

```diff
@@ -235,6 +236,31 @@ VTKImageImport< TOutputImage >
       }
     output->SetOrigin(outOrigin);
     }
+  if ( m_DirectionCallback )
+    {
+    double *inDirection = (m_DirectionCallback)( m_CallbackUserData );
+    typename TOutputImage::DirectionType outDirection;
+    for ( unsigned int i = 0; i < 3; ++i )
+      {
+      for ( unsigned int j = 0; j < 3; ++j )
+        {
+        if (i < OutputImageDimension && j < OutputImageDimension)
+          {
+          outDirection[i][j] = inDirection[i*3+j];
+          }
+        else if ((i != j) && (inDirection[i*3+j] != 0.0))
+          {
+            char ijk[4] = "IJK";
+            char xyz[4] = "XYZ";
+            itkExceptionMacro(<< "Can not convert a VTK image to an ITK image of dimension "
```

Comment:
This is probably the safe way to do it but it does raise the problem of backward compatibility. Maybe this exception should be surrounded by `#ifndef ITK_LEGACY_SILENT`. @thewtex : what do you think? Also, `can not` ->`cannot`

Reply:
I don't think we have to worry about backwards compatibility here because the *Direction* was never transferred.

Reply:
What I meant by backward compatibility is that someone who had some code working until now could see this exception appear

Reply:
So:
- `ITK_LEGACY_SILENT` to ensure we don't throw an exception if we didn't before
- another preprocessor definition based on the VTK version to avoid importing/exporting the direction altogether

Is that right?